homeassistant/components/overkiz/sensor.py

```diff
@@ -33,23 +34,39 @@
 from homeassistant.helpers.typing import StateType
 
 from . import HomeAssistantOverkizData
-from .const import DOMAIN, IGNORED_OVERKIZ_DEVICES, OVERKIZ_STATE_TO_TRANSLATION
+from .const import (
+    DOMAIN,
+    IGNORED_OVERKIZ_DEVICES,
+    OVERKIZ_STATE_TO_TRANSLATION,
+    OVERKIZ_UNIT_TO_HA,
+)
 from .coordinator import OverkizDataUpdateCoordinator
 from .entity import OverkizDescriptiveEntity, OverkizDeviceClass, OverkizEntity
 
 
+def _unit_of_measurement(device: Device, default: str) -> str | None:
+    """Retrieve the unit of measurement from Overkiz attributes."""
+    if unit := device.attributes.get(OverkizAttribute.CORE_MEASURED_VALUE_TYPE):
+        if unit.value_as_str:
+            return OVERKIZ_UNIT_TO_HA.get(unit.value_as_str, default)
+        else:
+            return default
+    return default
+
+
 @dataclass
 class OverkizSensorDescription(SensorEntityDescription):
     """Class to describe an Overkiz sensor."""
 
     native_value: Callable[[OverkizStateType], StateType] | None = None
+    native_unit_of_measurement: Callable[[Device], str | None] | None = None
```

Comment:
@iMicknl mypy complains on this

```
homeassistant/components/overkiz/sensor.py:62: error: Incompatible types in assignment (expression has type "Optional[Callable[[Device], Optional[str]]]", base class "SensorEntityDescription" defined the type as "Optional[str]")  [assignment]
Found 1 error in 1 file (checked 2 source files)
```

I don’t get where I’m wrong.

Reply:
I think its because `SensorEntityDescription` already has an attribute named `native_unit_of_measurement` which returns `Optional[str]`.

You should rename this attr to not overlap the base class attribute.

`get_unit_of_measurement`, `unit_of_measurement_fn` ?

Reply:
Thx for the hint, I will have a look.

Reply:
I was able to fix this issue. Thx 🙏